stdlib/iarray.ml

```diff
@@ -0,0 +1,339 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*             Xavier Leroy, projet Cristal, INRIA Rocquencourt           *)
+(*               Antal Spector-Zabusky, Jane Street, New York             *)
+(*                                                                        *)
+(*   Copyright 1996 Institut National de Recherche en Informatique et     *)
+(*     en Automatique.                                                    *)
+(*   Copyright 2023 Jane Street Group LLC                                 *)
```

Comment:
On the other hand, it might make sense to only keep this copyright line for the implementation.